src/app/(requires-auth)/management/branches/_hooks/useGetBranches.ts

```diff
@@ -0,0 +1,70 @@
+"use client";
+
+import { useEffect, useState, useTransition } from "react";
+import { z } from "zod";
+import { GetBranchesSchema } from "../_schemas";
+import { getBranches } from "../_actions";
+import { toast } from "sonner";
+import { Branch } from "@prisma/client";
+
+export const useGetBranches = () => {
+  const [isLoading, startTransition] = useTransition();
+  // ! how is the type generated by prisma?
+  const [branches, setBranches] = useState<Branch[] | null>(null);
+
+  /*
+TODO: Consider implementing optimistic UI updates to improve the perceived performance of the app. Update the UI optimistically as soon as the user initiates an action and then revert the changes if the action fails. This approach can make the application feel more responsive.
+
+TODO:
```

Comment:
This is a good suggestion, however, when it comes to testing, you'll have ask the question `What is the purpose of useGetBranches` , in this case I guess the answer will be to retrieve branches from an endpoint, and depending on the response will return branches or null, and a loading state. So what you'll be testing for is wether depending on the response state it returns an null or the branches, and wether the loading state behaves as expected. The reason for this tangent is having to pass getBranches into the hook, kind of negates the purpose of having a useGetBranches hook in the first place. If you were to pass getBrachhes into the hook then its better the hook was a generic one lets say `retrieveEntitiies` , where you can pass whatever api function (e.g getBranches, getStaff, e.tc) into it, in this case is clear that the purpose of such a hook is to retrieve entities (and not a specific entity), and as such needs to have an api function passed to it.